### PR TITLE
Use a single dockerfile for failtester images

### LINKED ISSUE #23: Unavailable postgres version
In failure testing docker files, the postgres version gives an error when building the image. I have fixed it by updating to the latest version which, at the time of creating this issue, is:
```
ENV PG_VERSION 11.5-1.pgdg90+1 
```

The related line:
https://github.com/citusdata/the-process/blob/master/circleci/images/failtester/Dockerfile-11#L80

### LINKED ISSUE #27: Failtester dockerfiles
For failtester image, we have 2 dockerfiles for postgres 10 and 11. These dockerfiles contain postgres dockerfile in them, which has the postgres version hardcoded in them. 

Instead of this, we can have a single dockerfile and take postgres version as an argument. We can use a postgres base image with that argument.

### PR DESCRIPTION
**Pull request by SaitTalhaNisanci**
2 Dockerfiles are replaced with only one, which uses postgres
as a base image with an argument that specifies the postgres major.
Python3.6 base image is replaced with its dockerfile.

To build the image in its dockerfile dir:

```bash
docker build --tag=<tag> . --build-arg PG_MAJOR=<pg-major>
```

Fixes #23.
Fixes #27.

### CODE REVIEW

File: circleci/images/failtester/files/sbin/install-testdeps [hunk 1]
```diff
@@ -20,23 +25,42 @@ apt-get install -y --no-install-recommends \
...
+curl -sf https://www.postgresql.org/media/keys/ACCC4CF8.asc | \
+APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=1 apt-key add -
+
+# add pgdg repo to sources
+echo "Writing /etc/apt/sources.list.d/pgdg.list..." >&2
+echo "deb http://apt.postgresql.org/pub/repos/apt/ ${codename}-pgdg main 12" > \
+     /etc/apt/sources.list.d/pgdg.list  
 
-pkg_cloud_url="https://packagecloud.io/install/repositories/citusdata/the-process/script.deb.sh"
+pkg_cloud_url="https://packagecloud.io/install/repositories/citus-bot/the-process/script.deb.sh"
```
Comment (JelteF) [inline circleci/images/failtester/files/sbin/install-testdeps]: why is this url changed? Same for below

File: circleci/images/failtester/files/sbin/install-testdeps [hunk 2]
```diff
@@ -20,23 +25,42 @@ apt-get install -y --no-install-recommends \
...
 
 echo "Installing tools for failure testing..." >&2
-pip install -Ir /tmp/etc/requirements.txt
+pip3 install -Ir /tmp/etc/requirements.txt
+
+# Give permission for usr so that circleci can write citus artifacts
+chmod o+w -R /usr/
+
+# install gosu so that we can run tests from root with circleci user
+apt-get update && apt-get install -y gosu
```
Comment (JelteF) [inline circleci/images/failtester/files/sbin/install-testdeps]: `apt-get update` is not needed here

File: circleci/images/failtester/files/sbin/install-testdeps (hunk 1 above)
Comment (SaitTalhaNisanci) [reply circleci/images/failtester/files/sbin/install-testdeps]: I dont have access to citusdata packagecloud, I only have access to citus-bot so I have uploaded the packages to citus-bot.

**Comment by JelteF** [comment]
Are the CircleCI tests still passing with these changes?

**Comment by SaitTalhaNisanci** [comment]
Yes this is a multi step process which is error prone. What I do is:
- Change the source code
- Build and push to docker hub
- Create a PR that uses the newly created images
- Make sure the tests pass
- Push the changes to this repository

[This PR](https://github.com/citusdata/citus/pull/3012) the images that are generated from this source code.

**Pull request merged**